src/courses/mappers/06.md

```diff
@@ -0,0 +1,309 @@
+---
+order: 6
+next: 07.md
+title: Formal OHDF Schema Definition
+author: Charles Hu
+---
+
+## Breaking Down the Formal OHDF Schema
+
+We can break down the formal OHDF schema in the same manner as before by observing each structure in the hierarchy from a top-down view.
+
+::: note Full OHDF Schema
+The following section contains a breakdown of a streamlined version of the OHDF schema. For the full and up-to-date version, see [here](https://saf.mitre.org/framework/normalize/ohdf-schema).
+:::
+
+### 1. <i>**Profiles**</i>
+
+::: details Profiles Schema
+```json
+profiles: [
+  0: {
+    name,
+    version,
+    sha256,
+    title,
+    maintainer,
+    summary,
+    license,
+    copyright,
+    copyright_email,
+    supports,
+    attributes,
+    groups,
+    controls,
+    status
+  },
+  ... // More items may exist if the security service produces multiple profiles per export
+]
+```
+:::
+
+- `name`: Name of profile, usually the original security service tool; should be unique.
+- `version`: Version of security service tool.
+- `sha256`: Checksum of the profile (NOTE: AUTOMATICALLY GENERATED BY HDF CONVERTERS, DO NOT POPULATE).
+- `title`: Title of security service scan; should be human readable.
+- `maintainer`: Maintainer.
+- `summary`: Summary of security service export (e.g., the STIG header).
+- `license`: Copyright license.
+- `copyright`: Copyright holder.
+- `copyright_email`: Copyright holder's email.
+- `supports`: Supported platform targets.
+- `attributes`: Inputs/attributes used in scan.
+- `groups`: Set of descriptions for the control groups (e.g., control IDs).
+- `controls`: Controls substructure (see below).
+- `status`: Status of profile (typically 'loaded').
+
+#### Example
+
+Here's an example of a <i>profiles</i> section from a converted GoSec data file:
+
+::: details OHDF Converted GoSec Data
+```json
+"profiles": [
+  {
+    "name": "Gosec scanner",
+    "title": "gosec",
+    "version": "dev",
+    "supports": [],
+    "attributes": [],
+    "groups": [],
+    "status": "loaded",
+    "controls": [
+      ...
+    ],
+    "sha256": "d0506f4b7715bf8b1cd81a8a87ab8efcce41ebeb2b5ec5fcfb23d3bdd136f48c"
+  }
+]
+```
+:::
+
+### 2. <i>**Controls**</i>
+
+::: details Controls Schema
+```json
+controls: [
+  0: {
+    id,
+    title,
+    desc,
+    descriptions,
+    impact,
+    refs,
+    tags,
+    code,
+    source_location,
+    results
+  },
+  ... // More items may exist if there are multiple controls reported per profile
+]
+```
+:::
+
+- `id`: ID of control, used for sorting; should be unique for each unique control.
+- `title`: Title of control.
+- `desc`: Description of the control.
+- `descriptions`: Additional descriptions; usually 'check' and 'fix' text for control.
```

Comment:
You might need to have some extended sections here and in other attributes to explain context.  Ex. we still have some inspec specific cruft we gotta deal with.  Or in this case, much of the guidance we work with is based off of stigs.  stigs specify check and fix text (explain what they are).  since we try to pass through all of the guidance, we have this generic descriptions section that can include that info along with any other info.